hw#15.py

```diff
@@ -0,0 +1,93 @@
+from enum import Enum
+from datetime import date
+import logging
+from logging.handlers import RotatingFileHandler
+
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(asctime)s - %(levelname)s - %(message)s")
+
+
+handler = RotatingFileHandler(
+    'user_actions.log',
+    maxBytes=1024,
+    backupCount=7,
+    encoding='utf-8'
+)
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+stream = logging.StreamHandler()
+stream.setFormatter(formatter)
+logger.addHandler(stream)
+
+
+class OrderStatus(Enum):
+    PENDING = 1
+    IN_PROGRESS = 2
+    READY = 3
+    COMPLETED = 4
+    CANCELLED = 5
+
+
+class Order:
+    _OrderStatus = iter(OrderStatus)
+    _data: dict[str, OrderStatus] = {}
+
+    def __init__(self, order_id: str):
+        self.order_id = order_id
+        self._data.setdefault(order_id, next(self._OrderStatus))
+
+    @classmethod
+    def update_status(cls, order_id: str):
+        if order_id in cls._data:
+            try:
+                cls._data[order_id] = next(cls._OrderStatus)
+            except StopIteration:
+                # Если достигнут конец последовательности статусов
+                logger.warning('Статус для заказа %s уже на последнем этапе.', order_id)
+        else:
+            logger.debug('Ошибка: заказ не найден.')
+
+    @classmethod
+    def display_status(cls, order_id: str):
+        if order_id in cls._data:
+            logger.info('Статус заказа %s: %s', order_id, cls._data[order_id].name)
+        else:
+            logger.error('Заказ с ID %s не найден.', order_id)
+
+
+def day_difference(first_date: str, second_date: str):
+    year, month, day = map(int, first_date.split())
```

Comment:
are you expecting the date in `11 11 2222` format?
A bit unusual format to me, better to ask user to enter `11.11.1234` and convert to to a day(I showed it on the lesson)